src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -32,33 +36,98 @@ public class TestJsonCommand : PSCmdlet
         /// then validates the JSON against the schema. Before testing the JSON string,
         /// the cmdlet parses the schema doing implicitly check the schema too.
         /// </summary>
-        [Parameter(Position = 1)]
+        [Parameter(Position = 1, ParameterSetName = TestJsonCommand.SchemaStringParameterSet)]
         [ValidateNotNullOrEmpty()]
         public string Schema { get; set; }
 
+        /// <summary>
+        /// A path to the file containg schema to validate the JSON against.
+        /// This is optional parameter.
+        /// If the parameter is absent the cmdlet only attempts to parse the JSON string.
+        /// If the parameter present the cmdlet attempts to parse the JSON string and
+        /// then validates the JSON against the schema. Before testing the JSON string,
+        /// the cmdlet parses the schema doing implicitly check the schema too.
+        /// </summary>
+        [Parameter(Position = 1, ParameterSetName = TestJsonCommand.SchemaFileParameterSet)]
+        [ValidateNotNullOrEmpty()]
+        public string SchemaFile { get; set; }
+
         private JsonSchema _jschema;
 
         /// <summary>
-        /// Prepare an JSON schema.
+        /// Process all exceptions in the AggregateException.
+        /// Unwrap TargetInvocationException if any and
+        /// rethrow inner exception without losing the stack trace.
+        /// </summary>
+        private static bool UnwrapException(Exception e)
+        {
+            if (e is TargetInvocationException)
+            {
+                ExceptionDispatchInfo.Capture(e.InnerException).Throw();
```

Comment:
I think we could avoid re-throw and write an error here.

Reply:
As [discussed below](https://github.com/PowerShell/PowerShell/pull/11934#discussion_r397142961), if schema is invalid (or not found) shouldn't we treat it as an invalid input and throw terminating error?

E.g.: `Test-Connection` will throw if it can't resolve host, `Test-ModuleManifest` throws if path is invalid or not pointing to the `psd1` file..